src/engine/quantizecontrol.cpp

```diff
@@ -84,28 +84,38 @@ double QuantizeControl::process(const double dRate,
     double prevBeat = m_pCOPrevBeat->get();
     double nextBeat = m_pCONextBeat->get();
     double closestBeat = m_pCOClosestBeat->get();
-    double currentClosestBeat = floor(m_pBeats->findClosestBeat(iCurrentSample));
 
-    if (closestBeat != currentClosestBeat) {
-        if (!even(static_cast<int>(currentClosestBeat))) {
-            currentClosestBeat--;
+    // Calculate this by hand since we may also want the beat locations themselves
+    // and duplicating the work would double the number of mutex locks.
+    double updatedPrev;
+    double updatedNext;
+    m_pBeats->findPrevNextBeats(iCurrentSample, &updatedPrev, &updatedNext);
+
+    if (updatedPrev == -1) {
+        if (updatedNext != -1) {
+            m_pCOClosestBeat->set(updatedNext);
+        } else {
+            // Likely no beat information -- can't set closest beat value.
```

Comment:
thinking about what to do here -- -1 isn't the right value to set because -1 is a valid sample